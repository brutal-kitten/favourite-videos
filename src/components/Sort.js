import React, { Component } from 'react';

class Sort extends Component {

  state = {
    value: 'no'
  }

  handleChange = (event) => {
    event.preventDefault();
    this.setState({value: event.target.value});
    this.props.sort(event.target.value);
  }
  render() {
    return(
      <div className="sort" >
        <span className="sortText">Sort by</span>
        <select id="select" value={this.state.value} onChange={(event) => this.handleChange(event)}>
          <option value="no">Not sorted</option>
          <option value="new">Newest added</option>
          <option value="old">Oldest added</option>
        </select>
      </div>
    )
  }
}

export default Sort;
