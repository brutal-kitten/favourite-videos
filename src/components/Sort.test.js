import React from 'react';
import ReactDOM from 'react-dom';
import Sort from './Sort.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders properly', () => {
  const tree = TestRenderer.create(
    <Sort />).toJSON();
  expect(tree).toMatchSnapshot();
});

  const mockSort = jest.fn();
  const props = {
    sort: mockSort
};

  const initialState = {
    value : 'new'
  }

  const wrapper = shallow(<Sort {...props}/>);

  it('initialize Sort with correct initial state', () => {
    expect(wrapper.state()).toEqual(initialState)
  });


describe('selection works properly', () => {

  const wrapper1 = shallow(<Sort {...props}/>);

  wrapper1.find('select').simulate('change', {
    target: {
      value: 'old'
    },
    preventDefault: () => {},
  });

  it('should call mockSort function with 1 given value', () => {
    expect(mockSort.mock.calls.length).toBe(1);
    expect(mockSort.mock.calls[0][0]).toBe("old");
  });

  it('updates value field in state', () => {
    expect(wrapper1.state()).toEqual({value: "old"})
  })
})
