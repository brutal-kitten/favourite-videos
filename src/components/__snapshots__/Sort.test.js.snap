// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="sort"
>
  <span
    className="sortText"
  >
    Sort by
  </span>
  <select
    id="select"
    onChange={[Function]}
    tabIndex="0"
    value="new"
  >
    <option
      value="new"
    >
      Newest added
    </option>
    <option
      value="old"
    >
      Oldest added
    </option>
  </select>
</div>
`;
